### PR TITLE
feat: segment migration of utils

### PR DESCRIPTION
**Pull request by NicolasMassart**
## **Description**

- Migrate utils
- move trackDappVisitedEvent to utils
- update unit tests
- update imports

## **Related issues**

Progresses https://github.com/MetaMask/mobile-planning/issues/1129

## **Manual testing steps**

NA
### **Before**

NA

### **After**

NA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by tommasini — DISMISSED** [review]
LGTM

File: app/util/middlewares.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import Logger from './Logger';
-import { trackErrorAsAnalytics } from './analyticsV2';
+import trackErrorAsAnalytics from './metrics/TrackError/trackErrorAsAnalytics';
```
Comment (Cal-L) [inline app/util/middlewares.js:2]: should this be replaced with `trackEvent`?

**Review by Cal-L — COMMENTED** [review]
Left one comment

File: app/util/middlewares.js (hunk 1 above)
Comment (NicolasMassart) [reply app/util/middlewares.js:2]: TrackErrorAsAnalytics is a util function that uses trackevent. We just moved it out of AnalyticsV2 to be able to remove it.

**Comment by NicolasMassart** [comment]
I will heck unit tests

File: app/components/Views/confirmations/components/MessageSign/index.test.tsx [hunk 2]
```diff
@@ -243,46 +235,6 @@ describe('MessageSign', () => {
...
-          chain_id: undefined,
-          signature_type: 'eth_sign',
-          security_alert_response: 'Benign',
-          security_alert_reason: '',
-          ppom_eth_chainId_count: 1,
-          version: undefined,
-        });
-      });
-    });
-
```
Comment (NicolasMassart) [inline app/components/Views/confirmations/components/MessageSign/index.test.tsx:285]: Tests are not unit test as they test implementation and side effects.
Also they are unpredicatable as they can't run separately and even before changes only worked when ran through the whole test suite.
This doesn't provide any security to the codebase and even give a false feeling of good coverage. This has to be reworked in the context of reworking all tests. For now, it's a blocker and really provides no value to be fixed. Removing testcase.

File: app/components/Views/confirmations/components/MessageSign/index.test.tsx [hunk 3]
```diff
@@ -361,30 +312,4 @@ describe('MessageSign', () => {
...
-        error: new Error(KEYSTONE_TX_CANCELED),
-      });
-      await waitFor(() => {
-        expect(analyticsV2.trackEvent).toHaveBeenCalledTimes(2);
-        expect((analyticsV2.trackEvent as jest.Mock).mock.calls[1][0]).toEqual(
-          MetaMetricsEvents.QR_HARDWARE_TRANSACTION_CANCELED,
-        );
-      });
-    });
-  });
```
Comment (NicolasMassart) [inline app/components/Views/confirmations/components/MessageSign/index.test.tsx:389]: Tests are not unit test as they test implementation and side effects.
Also they are unpredicatable as they can't run separately and even before changes only worked when ran through the whole test suite.
This doesn't provide any security to the codebase and even give a false feeling of good coverage. This has to be reworked in the context of reworking all tests. For now, it's a blocker and really provides no value to be fixed. Removing testcase.

**Review by tommasini — APPROVED** [review]
LGTM!

**Pull request merged**